internal/cloud/backend_plan.go

```diff
@@ -95,7 +87,25 @@ func (b *Cloud) opPlan(stopCtx, cancelCtx context.Context, op *backend.Operation
 		return nil, diags.Err()
 	}
 
-	return b.plan(stopCtx, cancelCtx, op, w)
+	// If the run errored, exit before checking whether to save a plan file
+	run, err := b.plan(stopCtx, cancelCtx, op, w)
+	if err != nil {
+		return run, err
```

Comment:
If we return an error, we should be make it clear that we are returning `nil, err`, I think.

Reply:
Hmm, yeah, all right